pydicom/valuerep.py

```diff
@@ -1248,6 +1248,48 @@ def DS(
     return DSfloat(val, auto_format, validation_mode)
 
 
+class ISfloat(float):
+    """Store value for an element with VR **IS** as :class:`float`.
+
+    Stores original integer string for exact rewriting of the string
+    originally read or stored.
+
+    Note: By the DICOM standard, IS can only be an :class:`int`,
+    however, it is not uncommon to see float IS values.  This class
+    is used if the config settings allow non-strict reading.
+
+    Generally, use :class:`~pydicom.valuerep.IS` to create IS values,
+    this is returned instead if the value cannot be represented as an
+    :class:`int`.  See :class:`~pydicom.valuerep.IS` for details of the
+    parameters and return values.
+    """
+    def __new__(  # type: ignore[misc]
+            cls: Type["ISfloat"], val: Union[str, float, Decimal],
+            validation_mode: int = None
+    ) -> float:
+        return super().__new__(cls, val)
+
+    def __init__(self, val: Union[str, float, Decimal],
+                 validation_mode: int = None) -> None:
+        # If a string passed, then store it
+        if isinstance(val, str):
+            self.original_string = val.strip()
+        elif isinstance(val, (IS, ISfloat)) and hasattr(val, 'original_string'):
```

Comment:
Reviewdog complains about this line that is 1 character too long. As long as we haven't changed this policy, this needs to be fixed... I'm not sure why codecov also complains, but I would say it got confused and we can ignore this.
@darcymason - I think we can merge this as soon as this minor formatting issue is fixed.

Reply:
> Reviewdog complains about this line that is 1 character too long. As long as we haven't changed this policy, this needs to be fixed

Looks like merge is not blocked, and this won't be a problem when we update the settings to allow longer lines, so I'm happy to leave it.



> I'm not sure why codecov also complains, but I would say it got confused and we can ignore this

That seems to happen quite frequently, at some point we should try to understand these differences, but for now I am happy to ignore.